test/end-to-end/core.sh

```diff
@@ -135,6 +135,48 @@ os::cmd::expect_success "docker pull ${DOCKER_REGISTRY}/cache/mysql:pullthrough"
 echo "[INFO] Docker start with GCS"
 os::cmd::expect_failure_and_text "docker run -e REGISTRY_STORAGE=\"gcs: {}\" openshift/origin-docker-registry:${TAG}" "No bucket parameter provided"
 
+# verify we can pull from tagged image (using tag)
+echo "[INFO] Removing ruby-22-centos7 images before next pull"
+imageid=$(docker images | grep centos/ruby-22-centos7 | awk '{ print $3 }')
+
+os::cmd::expect_success "docker rmi -f ${imageid}"
+echo "[INFO] Tagging ruby-22-centos7:latest to the same image stream and pulling it"
+os::cmd::expect_success "oc tag ruby-22-centos7:latest ruby-22-centos7:new-tag"
+os::cmd::expect_success "docker pull ${DOCKER_REGISTRY}/cache/ruby-22-centos7:new-tag"
+echo "[INFO] The same image stream pull successful"
+
+os::cmd::expect_success "docker rmi -f ${imageid}"
+echo "[INFO] Tagging ruby-22-centos7:latest to cross repository and pulling it"
+os::cmd::expect_success "oc tag ruby-22-centos7:latest cross:repo-pull"
+os::cmd::expect_success "docker pull ${DOCKER_REGISTRY}/cache/cross:repo-pull"
+echo "[INFO] Cross repository pull successful"
+
+os::cmd::expect_success "docker rmi -f ${imageid}"
+echo "[INFO] Tagging ruby-22-centos7:latest to cross namespace and pulling it"
+os::cmd::expect_success "oc tag cache/ruby-22-centos7:latest cross:namespace-pull -n custom"
+os::cmd::expect_success "docker pull ${DOCKER_REGISTRY}/custom/cross:namespace-pull"
+echo "[INFO] Cross namespace pull successful"
+
+# verify we can pull from tagged image (using imageid)
+tagid=$(oc get istag ruby-22-centos7:latest --template={{.image.metadata.name}})
+os::cmd::expect_success "docker rmi -f ${imageid}"
+echo "[INFO] Tagging ruby-22-centos7@${tagid} to the same image stream and pulling it"
+os::cmd::expect_success "oc tag ruby-22-centos7@${tagid} ruby-22-centos7:new-id-tag"
+os::cmd::expect_success "docker pull ${DOCKER_REGISTRY}/cache/ruby-22-centos7:new-id-tag"
+echo "[INFO] The same image stream pull successful"
+
+os::cmd::expect_success "docker rmi -f ${imageid}"
+echo "[INFO] Tagging ruby-22-centos7@${tagid} to cross repository and pulling it"
+os::cmd::expect_success "oc tag ruby-22-centos7@${tagid} cross:repo-pull-id"
+os::cmd::expect_success "docker pull ${DOCKER_REGISTRY}/cache/cross:repo-pull-id"
+echo "[INFO] Cross repository pull successful"
+
+os::cmd::expect_success "docker rmi -f ${imageid}"
+echo "[INFO] Tagging ruby-22-centos7@${tagid} to cross namespace and pulling it"
+os::cmd::expect_success "oc tag cache/ruby-22-centos7@${tagid} cross:namespace-pull-id -n custom"
+os::cmd::expect_success "docker pull ${DOCKER_REGISTRY}/custom/cross:namespace-pull-id"
+echo "[INFO] Cross namespace pull successful"
+
```

Comment:
Can we also add a test for pulling from chained cross repository? Also we need a test case for referencing image by dockerimage reference as well. Both can be done in a single test:

```
oc import-image --from="${DOCKER_REGISTRY}/custom/cross:namespace-pull" cross-cross:latest
docker pull ${DOCKER_REGISTRY}/cache/cross-cross:latest
```